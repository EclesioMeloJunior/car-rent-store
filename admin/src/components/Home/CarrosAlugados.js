import React, { useState } from "react";
import Col from "react-bootstrap/Col";
import Card from "react-bootstrap/Card";
import Badge from "react-bootstrap/Badge";
import Button from "react-bootstrap/Button";
import ButtonGroup from "react-bootstrap/ButtonGroup";

const defineStatus = status => {
	switch (status) {
		case "reservado": {
			return <Badge variant="dark">Reservado</Badge>;
		}
		case "pendente": {
			return <Badge variant="warning">Pendente</Badge>;
		}
		default:
			return <React.Fragment />;
	}
};

const CarrosAlugados = props => {
	const { carroAlugado } = props;

	return (
		<Col xs={12} lg={4} md={4}>
			<Card className="mt-3">
				<Card.Img
					style={{ height: "200px" }}
					variant="top"
					src="https://picsum.photos/500/500"
				/>
				<Card.Body>
					<small>#{carroAlugado.id}</small>
					<Card.Title>{carroAlugado.carro}</Card.Title>

					<div>
						<i class="fas fa-user mr-2"></i>
						<b>{carroAlugado.pessoa.nome}</b>
					</div>

					<div>
						<i class="fas fa-envelope mr-2"></i>
						<i>{carroAlugado.pessoa.email}</i>
					</div>

					<div>
						<i class="fas fa-phone mr-2"></i>
						<small>{carroAlugado.pessoa.telefone}</small>
					</div>
					<div className="mb-1">{defineStatus(carroAlugado.status)}</div>

					{carroAlugado.status === "pendente" && (
						<div>
							<ButtonGroup size="sm" aria-label="Carros Alugados Ações">
								<Button variant="success">Aceitar</Button>
								<Button variant="danger">Recusar</Button>
							</ButtonGroup>
						</div>
					)}

					{carroAlugado.status === "reservado" && (
						<div>
							<ButtonGroup size="sm" aria-label="Carros Alugados Ações">
								<Button variant="primary">Fechar</Button>
								<Button variant="secondary">Cancelar</Button>
							</ButtonGroup>
						</div>
					)}
				</Card.Body>

				<Card.Footer>
					<small className="text-muted">Período: 30/11/2019 - 03/12/2019</small>
				</Card.Footer>
			</Card>
		</Col>
	);
};

export default CarrosAlugados;
