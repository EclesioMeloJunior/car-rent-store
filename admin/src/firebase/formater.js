function fromSnapshotToArray(querySnapshot) {
	const normalArray = querySnapshot.docs.map(doc => {
		console.log(doc);
		return { ...doc.data(), id: doc.id };
	});

	return normalArray;
}

export { fromSnapshotToArray };
