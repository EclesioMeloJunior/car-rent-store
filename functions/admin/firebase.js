const admin = require("../admin");

async function uploadImageAndGetPath(imagePath, { bucket, metadata }) {
	const storageBucket = admin.storage().bucket(bucket);
	await storageBucket.upload(imagePath, {
		destination: imagePath,
		metadata: metadata
	});
}

module.exports = {
	uploadImageAndGetPath
};
