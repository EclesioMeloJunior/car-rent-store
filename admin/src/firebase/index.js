import * as firebase from "firebase";
import "firebase/auth";
import "firebase/firestore";
import "firebase/functions";

var firebaseConfig = {
  apiKey: "AIzaSyA4Vys1jo8XdpDYnnPBhfnNTn8dgHTk99g",
  authDomain: "car-rent-store.firebaseapp.com",
  databaseURL: "https://car-rent-store.firebaseio.com",
  projectId: "car-rent-store",
  storageBucket: "car-rent-store.appspot.com",
  messagingSenderId: "273341965801",
  appId: "1:273341965801:web:39342d81a1281b58"
};

firebase.initializeApp(firebaseConfig);

if (process.env.NODE_ENV === "development") {
  firebase.functions().useFunctionsEmulator("http://localhost:5000");
}

export default firebase;
