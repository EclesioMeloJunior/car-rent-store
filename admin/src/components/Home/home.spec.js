import React from "react";
import renderer from "react-test-renderer";
import CarrosAlugados from "./CarrosAlugados";
import firebase from "../../firebase";

const CARROS_ALUGADOS_DATA = [
	{
		id: 4,
		carro: "Ford Docus 3.0",
		pessoa: {
			nome: "Márcio Fernandez",
			email: "marcio@fernandez.com",
			telefone: "69 9 8105-0108"
		},
		checkin: {
			_seconds: 1,
			_nanoseconds: 1
		},
		checkout: {
			_seconds: 1,
			_nanoseconds: 1
		},
		status: "reservado"
	},
	{
		id: 4,
		carro: "Ford Docus 3.0",
		pessoa: {
			nome: "Márcio Fernandez",
			email: "marcio@fernandez.com",
			telefone: "69 9 8105-0108"
		},
		checkin: {
			_seconds: 1,
			_nanoseconds: 1
		},
		checkout: {
			_seconds: 1,
			_nanoseconds: 1
		},
		status: "reservado"
	}
];

describe("Quando buscar os carros alugados na API", () => {
	test("Deve renderizar as mesmas quantidades de carros na busca", () => {
		const carrosAlugadosComponents = renderer.create(
			CARROS_ALUGADOS_DATA.map((carroAlugado, carroAlugadoIndex) => (
				<CarrosAlugados key={carroAlugadoIndex} carroAlugado={carroAlugado} />
			))
		);

		const componentsTree = carrosAlugadosComponents.toJSON();

		expect(componentsTree.length).toBe(CARROS_ALUGADOS_DATA.length);
	});
});
