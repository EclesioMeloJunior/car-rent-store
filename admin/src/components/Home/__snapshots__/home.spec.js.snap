// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quando buscar os carros alugados na API Deve renderizar as mesmas quantidades de carros na busca 1`] = `
Array [
  <div
    className="col-lg-4 col-md-4 col-12"
  >
    <div
      className="mt-3 card"
    >
      <img
        className="card-img-top"
        src="https://picsum.photos/500/500"
        style={
          Object {
            "height": "200px",
          }
        }
      />
      <div
        className="card-body"
      >
        <small>
          #
          4
        </small>
        <div
          className="card-title h5"
        >
          Ford Docus 3.0
        </div>
        <div>
          <i
            class="fas fa-user mr-2"
          />
          <b>
            Márcio Fernandez
          </b>
        </div>
        <div>
          <i
            class="fas fa-envelope mr-2"
          />
          <i>
            marcio@fernandez.com
          </i>
        </div>
        <div>
          <i
            class="fas fa-phone mr-2"
          />
          <small>
            69 9 8105-0108
          </small>
        </div>
        <div
          className="mb-1"
        >
          <span
            className="badge badge-dark"
          >
            Reservado
          </span>
        </div>
        <div>
          <div
            aria-label="Carros Alugados Ações"
            className="btn-group btn-group-sm"
            role="group"
          >
            <button
              className="btn btn-primary"
              disabled={false}
              type="button"
            >
              Fechar
            </button>
            <button
              className="btn btn-secondary"
              disabled={false}
              type="button"
            >
              Cancelar
            </button>
          </div>
        </div>
      </div>
      <div
        className="card-footer"
      >
        <small
          className="text-muted"
        >
          Período: 30/11/2019 - 03/12/2019
        </small>
      </div>
    </div>
  </div>,
  <div
    className="col-lg-4 col-md-4 col-12"
  >
    <div
      className="mt-3 card"
    >
      <img
        className="card-img-top"
        src="https://picsum.photos/500/500"
        style={
          Object {
            "height": "200px",
          }
        }
      />
      <div
        className="card-body"
      >
        <small>
          #
          3
        </small>
        <div
          className="card-title h5"
        >
          Ford Docus 3.0
        </div>
        <div>
          <i
            class="fas fa-user mr-2"
          />
          <b>
            Márcio Fernandez
          </b>
        </div>
        <div>
          <i
            class="fas fa-envelope mr-2"
          />
          <i>
            marcio@fernandez.com
          </i>
        </div>
        <div>
          <i
            class="fas fa-phone mr-2"
          />
          <small>
            69 9 8105-0108
          </small>
        </div>
        <div
          className="mb-1"
        >
          <span
            className="badge badge-dark"
          >
            Reservado
          </span>
        </div>
        <div>
          <div
            aria-label="Carros Alugados Ações"
            className="btn-group btn-group-sm"
            role="group"
          >
            <button
              className="btn btn-primary"
              disabled={false}
              type="button"
            >
              Fechar
            </button>
            <button
              className="btn btn-secondary"
              disabled={false}
              type="button"
            >
              Cancelar
            </button>
          </div>
        </div>
      </div>
      <div
        className="card-footer"
      >
        <small
          className="text-muted"
        >
          Período: 30/11/2019 - 03/12/2019
        </small>
      </div>
    </div>
  </div>,
  <div
    className="col-lg-4 col-md-4 col-12"
  >
    <div
      className="mt-3 card"
    >
      <img
        className="card-img-top"
        src="https://picsum.photos/500/500"
        style={
          Object {
            "height": "200px",
          }
        }
      />
      <div
        className="card-body"
      >
        <small>
          #
          2
        </small>
        <div
          className="card-title h5"
        >
          Ford Docus 3.0
        </div>
        <div>
          <i
            class="fas fa-user mr-2"
          />
          <b>
            Márcio Fernandez
          </b>
        </div>
        <div>
          <i
            class="fas fa-envelope mr-2"
          />
          <i>
            marcio@fernandez.com
          </i>
        </div>
        <div>
          <i
            class="fas fa-phone mr-2"
          />
          <small>
            69 9 8105-0108
          </small>
        </div>
        <div
          className="mb-1"
        >
          <span
            className="badge badge-warning"
          >
            Pendente
          </span>
        </div>
        <div>
          <div
            aria-label="Carros Alugados Ações"
            className="btn-group btn-group-sm"
            role="group"
          >
            <button
              className="btn btn-success"
              disabled={false}
              type="button"
            >
              Aceitar
            </button>
            <button
              className="btn btn-danger"
              disabled={false}
              type="button"
            >
              Recusar
            </button>
          </div>
        </div>
      </div>
      <div
        className="card-footer"
      >
        <small
          className="text-muted"
        >
          Período: 30/11/2019 - 03/12/2019
        </small>
      </div>
    </div>
  </div>,
  <div
    className="col-lg-4 col-md-4 col-12"
  >
    <div
      className="mt-3 card"
    >
      <img
        className="card-img-top"
        src="https://picsum.photos/500/500"
        style={
          Object {
            "height": "200px",
          }
        }
      />
      <div
        className="card-body"
      >
        <small>
          #
          1
        </small>
        <div
          className="card-title h5"
        >
          Ford Docus 3.0
        </div>
        <div>
          <i
            class="fas fa-user mr-2"
          />
          <b>
            Márcio Fernandez
          </b>
        </div>
        <div>
          <i
            class="fas fa-envelope mr-2"
          />
          <i>
            marcio@fernandez.com
          </i>
        </div>
        <div>
          <i
            class="fas fa-phone mr-2"
          />
          <small>
            69 9 8105-0108
          </small>
        </div>
        <div
          className="mb-1"
        >
          <span
            className="badge badge-dark"
          >
            Reservado
          </span>
        </div>
        <div>
          <div
            aria-label="Carros Alugados Ações"
            className="btn-group btn-group-sm"
            role="group"
          >
            <button
              className="btn btn-primary"
              disabled={false}
              type="button"
            >
              Fechar
            </button>
            <button
              className="btn btn-secondary"
              disabled={false}
              type="button"
            >
              Cancelar
            </button>
          </div>
        </div>
      </div>
      <div
        className="card-footer"
      >
        <small
          className="text-muted"
        >
          Período: 30/11/2019 - 03/12/2019
        </small>
      </div>
    </div>
  </div>,
]
`;
